CorsixTH/Lua/app.lua

```diff
@@ -616,9 +640,8 @@ end
 --!return (string, error) Returns the found absolute path, or nil if not found. Then
 --!       a second variable is returned with an error message.
 function App:getAbsolutePathToLevelFile(level)
-  local path = debug.getinfo(1, "S").source:sub(2, -12)
   -- First look in Campaigns. If not found there, fall back to Levels.
-  local list_of_possible_paths = { self.user_level_dir, path .. "Campaigns", self.level_dir }
+  local list_of_possible_paths = { self.user_campaign_dir, self.campaign_dir, self.user_level_dir, self.level_dir }
```

Comment:
Not your fault, but yuck.

"list_of" is type information that is already expressed at the right of the `=` sign. Also it's clear the right side that it's a list of paths, the name doesn't need to tell that too. The reader now has to check what happens with the variable to understand why it exists.

Instead, use it to express the reason of having the variable. Something like "paths_to_search" says what the intention of that data is. From that intention it's fully logical that you iterate over the values next.